src/util/exportRequestBuilder.ts

```diff
@@ -0,0 +1,66 @@
+import { SupportedExportTypes } from '@/components/query-selector/export-type';
+import { TypeElement } from '@/state/type-element-params-state';
+
+/*
+ * Parameters to build export request string.
+ */
+export interface BuilderRequest {
+  baseUrl: string;
+  exportType: SupportedExportTypes;
+  queryParams: BuilderRequestQueryParams;
+  id?: string;
+}
+
+/*
+ * Query parameters that modify the export request.
+ */
+export interface BuilderRequestQueryParams {
+  type: string[];
+  element: string[];
+  typeElement: TypeElement[];
+  typeFilter?: string[]; // may not be string[] in the future
+}
+
+/*
+ * Builds bulk export request string from a list of parameters, a SupportedExportType, and an id.
+ * Note: currently only supports type parameters and elements.
+ */
+export function buildExportRequestString(request: BuilderRequest) {
+  const { baseUrl, exportType, id, queryParams } = request;
+
+  const exportPath = buildExportPath(exportType, id);
+
+  const queryString = buildQueryString(queryParams) ?? '';
```

Comment:
```suggestion
  const { baseUrl, exportType, id, queryParams } = request;
  const exportPath = buildExportPath(exportType, id);
  const queryString = buildQueryString(queryParams) ?? '';
```
Lol I have such a hard time deciding how much white space is too much...I feel like this makes a bit more sense and matches the general vibe of the rest of the file. Nitpicky but it was something I immediately noticed